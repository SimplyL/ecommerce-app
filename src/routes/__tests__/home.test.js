import React from 'react';
import { shallow } from 'enzyme';
import Home from '../home';
import mockProducts from './__mocks__/products';

describe('Home page', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Home products={mockProducts} />);
  });

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render product list with correct item number', () => {
    expect(wrapper.find('ProductItem')).toHaveLength(2);
  });
});
