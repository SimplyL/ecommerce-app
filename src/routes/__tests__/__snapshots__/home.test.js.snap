// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render without crashing 1`] = `
<Fragment>
  <Title
    name="Browse products"
  />
  <ProductList>
    <ProductItem
      key="0"
      product={
        Object {
          "description": "Test description",
          "id": "0",
          "image": "roxy-sugar-banana.jpg",
          "name": "Model",
          "price": 4.99,
        }
      }
    />
    <ProductItem
      key="1"
      product={
        Object {
          "description": "Test",
          "id": "1",
          "image": "gnu-fb-head.jpg",
          "name": "Model",
          "price": 49.99,
        }
      }
    />
  </ProductList>
</Fragment>
`;
